src/storage/mod.rs

```diff
@@ -488,15 +488,35 @@ impl<E: Engine> TestStorageBuilder<E> {
     }
 }
 
+/// `Storage` implements transactional KV APIs and raw KV APIs on a given `Engine`. An `Engine`
+/// provides low level KV functionality. `Engine` has multiple implementations. When a TiKV server
+/// is running, a `RaftKv` will be the underlying `Engine` of `Storage`. The other two types of
+/// engines are for test purpose.
+///
+/// `Storage` is actually a reference to storage. It is cloneable, and a clone of a `Storage` is
```

Comment:
Seems to be too verbose. How about just: `Storage` is reference counted and cloning `Storage` will just increase the reference counter. Storage resources (i.e. threads, engine) will be released when all references are dropped.